inline CreateConfigMap into the places where it is used

anishbista60 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as draft: https://github.com/kubevirt/kubevirt/blob/main/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
### Description
Inlined ```CreateConfigMap``` present in `tests/utils.go`  into the places where it is used .
### What this PR does
Before this PR:

The functions `CreateConfigMap` in `tests/utils.go` doesnot look so useful because these function being called again and again in various file.

After this PR:

 This function is removed from `tests/utils.go` and inlined into the place where they are used.


<!-- (optional, in `fixes #11436(, fixes #11436, ...)` format, will close the issue(s) when PR gets merged)*: -->
Fixes #11436

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: <!-- https://github.com/kubevirt/kubevirt/issues/11436 ... -->

### Special notes for your reviewer

<!-- optional -->

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
None
```

anishbista60 commented:
@0xFelix  please look over this changes

0xFelix commented:
That looks good so far. Only one nit, could you put something into the body of the commit message please? Something similar to what you have written in the PR description. Thanks!

Let's see what CI says
/test all

0xFelix commented:
@anishbista60 You need to rebase this PR before it can be merged, thanks.

anishbista60 commented:
@0xFelix Except squash commit . Is here any extra changes to be done ?

0xFelix commented:
Same comment as on #11450

anishbista60 commented:
@0xFelix Done sir and thankyou for guiding me.

0xFelix commented:
Sure and same comment as on #11450

anishbista60 commented:
@0xFelix  over here also

0xFelix commented on tests/config_test.go:
```diff
@@ -35,15 +36,18 @@ import (
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
 	"kubevirt.io/kubevirt/tests/libvmi"
 	"kubevirt.io/kubevirt/tests/testsuite"
+	"kubevirt.io/kubevirt/tests/libpod"
```
Was this reordered by `make format`?

anishbista60 replied:
@0xFelix  no sorry i did mistake while resloving merge conflict . shall i interchanged the /lipod and /kubevirt package.

anishbista60 replied:
@0xFelix  Done

0xFelix reviewed (commented):
One comment, otherwise looks good to me.

Can you give it a look too @alicefr?

anishbista60 commented:
@alicefr sure, please look over it .

alicefr commented:
Looks good!
/approve

0xFelix reviewed (approved):
Thanks!

/lgtm

alicefr commented:
@anishbista60 please fix the build failures. The files should be modified after the compilation. See `pull-kubevirt-build` CI job

anishbista60 commented:
i think i did one big mistake , when i cloned the repo. `git status` was showing 
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-ar
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-cpp
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-gcc
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-gcov
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-ld
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-nm
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-objdump
        modified:   bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-strip
        
        These changes are automatically happen so that  i discard this changes 
        
        with `git checkout --` 
        That's why job are failing
        
        Now, Can i retrive these changes ? How to solve that problem ?

0xFelix commented:
Run `make format` and amend the changes to your commit. That should solve the issue.

anishbista60 commented:
@0xFelix  shall i  format it from root  or /bazel folder.

anishbista60 commented:
@0xFelix   I run the make fomat from /kubervirt directory.

anishbista60 commented:
@0xFelix i'm scared looking what's happening

anishbista60 commented:
@0xFelix could you please checkout this changes

alicefr commented on bazel/toolchain/aarch64-none-linux-gnu/wrappers/aarch64-none-linux-gnu-ar:
```diff
@@ -1 +0,0 @@
-wrapper
```
Please, remove all those changes

anishbista60 replied:
@alicefr  i have made changes on the recommdation of @0xFelix( Run make format and amend the changes to your commit. That should solve the issue.)

alicefr commented:
/hold
@anishbista60 please remove all the changes to the aarch64 files

0xFelix commented:
/test all

anishbista60 commented:
@alicefr  but the working directory is clean . Why jobs are failing?

0xFelix commented:
Your formatting is off. You need to run `make format` at the project's root. Then you need to amend the changes to your commit.

anishbista60 commented:
> Your formatting is off. You need to run `make format` at the project's root. Then you need to amend the changes to your commit.

Is this a cause for failing of job. 
for format shall i run for main or inline-02 branch?

0xFelix commented:
You need to format your changes in `inline-02`.

anishbista60 commented:
@0xFelix yeah done sir , Run the `make format` from /kubevirt inline-02 branch  and ammed the changes .please check it out

anishbista60 commented:
@0xFelix sorry for push  , got confused with  changes made by `make format` and finally got to know that it was made by`make format`  and again rebase to previous changes.

alicefr commented:
/test all

anishbista60 commented:
@0xFelix  which changes do i need to take ? got confused

0xFelix commented on tests/config_test.go:
```diff
@@ -145,7 +153,12 @@ var _ = Describe("[rfe_id:899][crit:medium][vendor:cnv-qe@redhat.com][level:comp
+						Data:       map[string]string{"option": "value"},
+					}, metav1.CreateOptions{})
+
+					Expect(err).NotTo(HaveOccurred())
```
I'd swap order of empty line in Expect, so Expect comes immediately after the assignment to `err`, and then a new line.

anishbista60 replied:
@0xFelix should make changes to all file or just in `config_test.go`

0xFelix reviewed (commented):
@anishbista60 You need to rebase your PR on main again, so it can be merged.

victortoso commented:
@anishbista60 how is it going with this one? Please remember to remove the merge commit. You shouldn't have merge commits in the PR, just the commits you want to add in the `main` branch.

anishbista60 commented:
@victortoso yeah , going to Modify it

victortoso commented:
/test pull-kubevirt-build
/test pull-kubevirt-generate

victortoso commented on tests/config_test.go:
```diff
@@ -356,7 +366,11 @@ var _ = Describe("[rfe_id:899][crit:medium][vendor:cnv-qe@redhat.com][level:comp
+					ObjectMeta: metav1.ObjectMeta{Name: configMapName},
+					Data:       configData,
+				}, metav1.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
```
Just random statistics:

```
toso@tapioca ~/s/k/kubevirt (main)> grep -rniI "NotTo.Have" tests/ | wc -l
270
toso@tapioca ~/s/k/kubevirt (main)> grep -rniI "ToNot.Have" tests/ | wc -l
2487
```

So, `ToNot` is used about 10 times more in `tests/` than `NotTo`.

anishbista60 replied:
@victortoso Done sir

victortoso reviewed (changes requested):
*pull-kubevirt-build* fails:
```
 Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   tests/config_test.go
	modified:   tests/migration/migration.go
	modified:   tests/operator/operator.go
	modified:   tests/sysprep_test.go
no changes added to commit (use "git add" and/or "git commit -a") 
```

running `make format` shows:
```
On branch review-11469
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   tests/config_test.go
        modified:   tests/migration/migration.go
        modified:   tests/operator/operator.go
        modified:   tests/sysprep_test.go

no changes added to commit (use "git add" and/or "git commit -a")
```

Please fix the issues locally in your machine before asking for another review.

0xFelix commented:
/test all

anishbista60 commented:
@0xFelix I check the log but unable to figure out . could please suggest how to troubleshoot? OR do i need to run `make generate`

victortoso commented:
/test pull-kubevirt-e2e-k8s-1.29-sig-storage
/test pull-kubevirt-e2e-arm64

>  I check the log but unable to figure out 

Just some unrelated issues, it seems. In the `sig-storage` we have `admission webhook "datavolume-mutate.cdi.kubevirt.io" denied the request: etcdserver: request timed out`. `e2e-arm64` shows a `panic` due lack of memory.

Please have pacience, your PR will be reviewed and if anything else is pending, we'll let you know.

anishbista60 commented:
@victortoso if you don't mind , could you please re run `pull-kubevirt-check-tests-for-flake`

anishbista60 commented:
@0xFelix could you please re run `pull-kubevirt-check-tests-for-flakes`

0xFelix commented:
/test pull-kubevirt-check-tests-for-flakes

alicefr commented:
/test pull-kubevirt-check-tests-for-flakes

anishbista60 commented:
@alicefr unable to figure out the reason behind it . Could you please suggest?

anishbista60 commented:
@0xFelix could you please test it ?

0xFelix commented:
Sure!

/test all

anishbista60 commented:
@0xFelix every thing looks good but why job are failing. could you please assist me  and re Run the failed case .

0xFelix commented:
@anishbista60 Looks like flakes. Retesting.

/test pull-kubevirt-conformance-arm64
/test pull-kubevirt-e2e-k8s-1.29-sig-storage
/test pull-kubevirt-check-tests-for-flakes

anishbista60 commented:
@0xFelix please remove the `hold` label.

0xFelix commented on tests/config_test.go:
```diff
@@ -34,6 +34,8 @@ import (
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	"golang.org/x/crypto/ssh"
+
```
nit: Would you mind removing this empty line?

0xFelix reviewed (commented):
/hold cancel

anishbista60 commented:
@0xFelix remove the extra space. Now , i hope flakes job will pass .

0xFelix commented:
/test all

anishbista60 commented:
@0xFelix now it's enough Seeing `flakes ` getting failed again and again . It's code fault or testing issue .

0xFelix commented:
@anishbista60 Not your fault.

/test pull-kubevirt-check-tests-for-flakes

anishbista60 commented:
@0xFelix no way Again failed .

0xFelix commented:
Looks to me like the lane is running longer than its definition permits.

@dhiller Can you help with this?

anishbista60 commented:
@0xFelix looks this Issue is already merged on someone else PR  .

dhiller commented:
> Looks to me like the lane is running longer than its definition permits.
> 
> @dhiller Can you help with this?

@0xFelix hm, I see that the lane is trying to run 750 test cases, among those migration and operator tests. This is an edge case where not only large test files containing many tests have been changed but also those tests take a long time to run.

Honestly, I would not want to extend the time this lane runs to more than it is allowed to now. Instead we should work to split up the test files that have many tests into segments so that the number of tests inside a test file becomes smaller.

This would in many cases help reduce the number of tests being run on the check-test-for-flakes lane. A positive side effect of this would be making it easier for people coming into the project understand what the tests do.

WDYT?
